packaging/setup/plugins/ovirt-engine-common/ovirt-engine-keycloak/core/admin_user.py

```diff
@@ -98,19 +122,17 @@ def _create_admin(self):
                     ]
                 },
             )
-            os.chown(
-                okkcons.FileLocations.KEYCLOAK_ADD_INITIAL_ADMIN_FILE,
-                osetuputil.getUid(
-                    self.environment[osetupcons.SystemEnv.USER_ENGINE],
-                ),
-                osetuputil.getGid(
-                    self.environment[osetupcons.SystemEnv.GROUP_ENGINE],
-                ),
+
+            new_user_json = os.path.join(
+                self._tempDir.directory,
+                okkcons.Const.KEYCLOAK_ADD_USER_JSON,
             )
-            os.chmod(
-                okkcons.FileLocations.KEYCLOAK_ADD_INITIAL_ADMIN_FILE,
-                0o600
+
+            self._set_permissions(
+                new_user_json=new_user_json,
+                environment=self.environment,
             )
+            os.rename(new_user_json, okkcons.FileLocations.KEYCLOAK_ADD_INITIAL_ADMIN_FILE)
```

Comment:
python docs say:
```
 The operation may fail on some Unix flavors if src and dst are on different filesystems. If successful, the renaming will be an atomic operation (this is a POSIX requirement).
```
So it might be better to create your temporary dir inside OVIRT_ENGINE_CONFIG_DIR, to try and make sure  they are on the same filesystem. You'll not be able to user service.TempDir because it does not allow setting the parent ('dir' arg of mkdtemp). If it worked for you in OST, it might mean we are "not affected" - it has /tmp on a separate filesystem. But better safe than sorry.

Reply:
Ack  - updating.